vidarr-server/src/main/java/ca/on/oicr/gsi/vidarr/server/PriorityByWorkflow.java

```diff
@@ -0,0 +1,147 @@
+package ca.on.oicr.gsi.vidarr.server;
+
+import static java.util.Map.Entry.comparingByValue;
+
+import ca.on.oicr.gsi.Pair;
+import ca.on.oicr.gsi.vidarr.BasicType;
+import ca.on.oicr.gsi.vidarr.ConsumableResource;
+import ca.on.oicr.gsi.vidarr.ConsumableResourceResponse;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.prometheus.client.Gauge;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.SortedSet;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListSet;
+
+
+final class PriorityByWorkflow implements ConsumableResource {
+
+  private List acceptedPriorities = Arrays.asList(1, 2, 3, 4);
+
+  private static final class WaitingState {
+
+    private final SortedSet<SimpleEntry<String, Integer>> waiting = new ConcurrentSkipListSet<SimpleEntry<String, Integer>>(comparingByValue()){
+
+      @Override
+      public boolean add(SimpleEntry<String, Integer> simpleEntry) {
+        super.remove(this.getByKey(simpleEntry.getKey()));
```

Comment:
`remove` only if the results of `this.getByKey` is not null

Reply:
bump

Reply:
super.remove here "Removes the specified element from this set if it is present"

Reply:
Error on startup:
```
Exception in thread "main" java.lang.NullPointerException
     at java.base/java.util.concurrent.ConcurrentSkipListMap.remove(ConcurrentSkipListMap.java:1799)
     at java.base/java.util.concurrent.ConcurrentSkipListSet.remove(ConcurrentSkipListSet.java:261)
     at ca.on.oicr.gsi.vidarr.server@0.12.1-SNAPSHOT/ca.on.oicr.gsi.vidarr.server.PriorityByWorkflow$$WaitingState$1.add(PriorityByWorkflow.java:31)
```
`ConcurrentSkipListSet`'s `remove` function calls `ConcurrentSkipListMap`'s `remove` function. This is ConcurrentSkipListMap's `remove` function, as viewed in my IDE:
```
public boolean remove(Object key, Object value) {
    if (key == null)
        throw new NullPointerException();
    return value != null && doRemove(key, value) != null;
}   
```
I really do think there needs to be a null check before calling `remove`